libraries/Microsoft.Bot.Connector/Authentication/BotFrameworkAuthentication.cs

```diff
@@ -49,6 +50,12 @@ public abstract class BotFrameworkAuthentication
         /// <returns>Asynchronous Task with <see cref="UserTokenClient" /> instance.</returns>
         public abstract Task<UserTokenClient> CreateUserTokenClientAsync(ClaimsIdentity claimsIdentity, CancellationToken cancellationToken);
 
+        /// <summary>
+        /// Creates a <see cref="BotFrameworkClient"/> used for calling Skills.
+        /// </summary>
+        /// <returns>A <see cref="BotFrameworkClient"/> instance to call Skills.</returns>
+        public abstract BotFrameworkClient CreateBotFrameworkClient();
```

Comment:
Seems like [this class already shipped](https://github.com/microsoft/botbuilder-dotnet/blob/4.12/libraries/Microsoft.Bot.Connector/Authentication/BotFrameworkAuthentication.cs), so we cannot really add this abstract method. Any customer who inherited from this class would get broken by this new abstract method. You could make it virtual to come around this.

Reply:
opps, sorry meant to provide implementation, I'll add next commit